pkg/handler/project/project.go

```diff
@@ -770,6 +690,139 @@ func (h *handler) UpdateMember(c *gin.Context) {
 	c.JSON(http.StatusOK, view.CreateResponse(view.ToCreateMemberData(slot), nil, done(nil), nil, ""))
 }
 
+func (h *handler) assignMemberToProject(db *gorm.DB, slotID string, projectID string, input UpdateMemberInput) (*model.ProjectMember, error) {
+	// check is slot contains any member?
+	member, err := h.store.ProjectMember.GetBySlotID(db, slotID)
+	if err != nil && !errors.Is(err, gorm.ErrRecordNotFound) {
+		h.logger.Fields(logger.Fields{"slotID": slotID}).Error(err, "failed to get project member by slotID")
+		return nil, err
+	}
+
+	if member.Status == model.ProjectMemberStatusInactive {
+		h.logger.Fields(logger.Fields{"member": member}).Error(ErrSlotIsInactive, "slot is inactive")
+		return nil, ErrSlotIsInactive
+	}
+	if member.EmployeeID != input.EmployeeID {
+		h.logger.
+			Fields(logger.Fields{"member": member}).
+			Error(ErrSlotAlreadyContainsAnotherMember, "slot already contains another member")
+		return nil, ErrSlotAlreadyContainsAnotherMember
+	}
+
+	// check is member active in project?
+	member, err = h.store.ProjectMember.One(db, projectID, input.EmployeeID.String(), model.ProjectMemberStatusActive.String())
```

Comment:
why the function name is `One()` but filter `status` also. Should be `GetOneByStatus()`. Or keep `One()` is by `project_id = ? AND employee_id = ?` and check `status` outside